### PR TITLE
test(NNS): run upgrade_canisters_with_golden_nns_state daily

### PR DESCRIPTION
**Pull request by basvandijk**
This adds the new job `nns-tests-nightly` to the `schedule-daily` workflow which runs the
`//rs/nns/integration_tests:upgrade_canisters_with_golden_nns_state` rust test.

### CODE REVIEW

**Review by marko-k0 — APPROVED** [review]

File: .github/workflows/schedule-daily.yml
Comment (daniel-wong-dfinity-org) [inline .github/workflows/schedule-daily.yml:114]: upgrade_canisters_with_godlen_nns_state only takes 5 minutes on my devenv VM. If this is the only test being run here, I'd suggest reducing this to something like 2-5x (i.e. 10 - 25 min). But I guess the idea is that more tests would be added (in rs/nns with tage nns_tests_nightly) later? Currently, there are none (but ofc, this PR adds the tag to a test).

File: rs/nns/integration_tests/BUILD.bazel
Comment (daniel-wong-dfinity-org) [inline rs/nns/integration_tests/BUILD.bazel:394]: I would not mind this being run more frequently (e.g. hourly). FWIW, it only takes 5 minutes (on my devenv vm). Ofc, it does use a fair amount of disk (10 GB or so).

File: rs/nns/integration_tests/src/upgrade_canisters_with_golden_nns_state.rs
Comment (daniel-wong-dfinity-org) [inline rs/nns/integration_tests/src/upgrade_canisters_with_golden_nns_state.rs:19]: Hmm. I am surprised clippy did not block this from being submitted before.

**Review by daniel-wong-dfinity-org — APPROVED** [review]
Thank you for doing this! Been meaning to get this set up.

Can I request that in a future PR, we have a Slack message sent to dev-nns when a run fails?

**Comment by daniel-wong-dfinity-org** [comment]
In the future, we may want automatic retry for flakes. upgrade_canisters_with_golden_nns_state seems to be very not flakey. Therefore, I do not think we need this **_yet_**. Would this just a matter of (adding|modifying) a single line of configuration?

**Comment by daniel-wong-dfinity-org** [comment]
Instead of saying `chore`, in the title, I would suggest maybe `test`.

**Comment by basvandijk** [comment]
@daniel-wong-dfinity-org 
> Can I request that in a future PR, we have a Slack message sent to dev-nns when a run fails? 

That seems like a nice addition. There could be a step [like this](https://github.com/dfinity/ic/blob/master/.github/workflows/ci-notify-slack.yml#L31-L39)  after the `bazel-test-all` step.  `bazel-test-all` will then need to enable [continue-on-error](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) to ensure the job doesn't fail before it sends the slack notification.

This is also something the NNS team can add themselves after this merges.

File: rs/nns/integration_tests/BUILD.bazel
Comment (basvandijk) [reply rs/nns/integration_tests/BUILD.bazel:394]: Let's first try this out daily and if that proves stable we move it to hourly.

File: rs/nns/integration_tests/src/upgrade_canisters_with_golden_nns_state.rs
Comment (basvandijk) [reply rs/nns/integration_tests/src/upgrade_canisters_with_golden_nns_state.rs:19]: I'm surprised as well!

File: .github/workflows/schedule-daily.yml
Comment (basvandijk) [reply .github/workflows/schedule-daily.yml]: Done.

**Pull request merged**
